src/go/rpk/pkg/cli/transform/deploy.go

```diff
@@ -233,12 +240,16 @@ func mergeProjectConfigs(lhs project.Config, rhs project.Config) (out project.Co
 	if len(rhs.OutputTopics) > 0 {
 		out.OutputTopics = rhs.OutputTopics
 	}
+	if rhs.Compression != "" {
```

Comment:
This is why the flag should default to empty string 😄

Reply:
Yeah, that's what I was harping on [up here](https://github.com/redpanda-data/redpanda/pull/18514#discussion_r1625005753) 🤷 

I have to push again to fix a conflict, so I think I'll change it back. But I (clearly) don't have a strong opinion about it.

Reply:
Yeah I was agreeing with you 😄